components/camel-xj/src/main/java/org/apache/camel/component/xj/XsltBuilder.java

```diff
@@ -0,0 +1,482 @@
+import org.slf4j.LoggerFactory;
+import org.xml.sax.EntityResolver;
+
+import javax.xml.transform.*;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.transform.stream.StreamSource;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import static org.apache.camel.util.ObjectHelper.notNull;
+
+// todo?: This class is just a copy from the XsltComponent because of minor but important different requirements. Refactoring of XsltComponent so that we can extend it?
+
+/**
+ * Creates a <a href="http://camel.apache.org/processor.html">Processor</a>
+ * which performs an XSLT transformation of the IN message body.
+ * <p/>
+ * Will by default output the result as a String. You can chose which kind of output
+ * you want using the <tt>outputXXX</tt> methods.
+ */
+public class XsltBuilder implements Processor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(XsltBuilder.class);
+    private Map<String, Object> parameters = new HashMap<>();
+    private XMLConverterHelper converter = new XMLConverterHelper();
+    private Templates template;
+    private volatile BlockingQueue<Transformer> transformers;
+    private SourceHandlerFactory sourceHandlerFactory;
+    private ResultHandlerFactory resultHandlerFactory = new StringResultHandlerFactory();
+    private boolean failOnNullBody = true;
+    private URIResolver uriResolver;
+    private boolean deleteOutputFile;
+    private ErrorListener errorListener;
+    private boolean allowStAX = true;
+    private EntityResolver entityResolver;
+
+    public XsltBuilder() {
+    }
+
+    public XsltBuilder(Templates templates) {
+        this.template = templates;
+    }
+
+    @Override
+    public String toString() {
+        return "XSLT[" + template + "]";
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        notNull(getTemplate(), "template");
+
+        if (isDeleteOutputFile()) {
+            // add on completion so we can delete the file when the Exchange is done
+            String fileName = ExchangeHelper.getMandatoryHeader(exchange, Exchange.XSLT_FILE_NAME, String.class);
+            exchange.addOnCompletion(new XsltBuilderOnCompletion(fileName));
+        }
+
+        Transformer transformer = getTransformer();
+        configureTransformer(transformer, exchange);
+
+        ResultHandler resultHandler = resultHandlerFactory.createResult(exchange);
+        Result result = resultHandler.getResult();
+        // let's copy the headers before we invoke the transform in case they modify them
+        Message out = exchange.getOut();
+        out.copyFrom(exchange.getIn());
+
+        // the underlying input stream, which we need to close to avoid locking files or other resources
+        InputStream is = null;
```

Comment:
This is just initialized to null and then is this null closed. This doesnt seem correct.

Reply:
obsolete.